src/test/java/org/xerial/snappy/SnappyOutputStreamTest.java

```diff
@@ -163,8 +163,10 @@ public void batchingOfWritesShouldNotAffectCompressedDataSize()
         // Compress the data once so that we know the expected size:
         byte[] expectedCompressedData = compressAsChunks(orig, Integer.MAX_VALUE);
         // Hardcoding an expected compressed size here will catch regressions that lower the
-        // compression quality:
-        assertEquals(91013, expectedCompressedData.length);
+        // compression quality. On little-endian platforms, the expected size is 91013; on
+        // big-endian platforms, the expected size is 90943.
+        assertTrue(String.format("unexpected compressed data size (%d)", expectedCompressedData.length),
```

Comment:
You can use `ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN` to check big-endian platform.